trees/redblacktree/redblacktree.go

```diff
@@ -192,6 +203,26 @@ func (tree *Tree) Right() *Node {
 	return parent
 }
 
+// Min returns the minimum key value pair of the AVL tree
+// or nils if the tree is empty.
+func (tree *Tree) Min() (interface{}, interface{}) {
+	n := tree.Left()
+	if n == nil {
+		return nil, nil
+	}
+	return n.Key, n.Value
+}
+
+// Max returns the minimum key value pair of the AVL tree
+// or nils if the tree is empty.
+func (tree *Tree) Max() (interface{}, interface{}) {
```

Comment:
Have a good night! Let me know what you want to do with keeping the abstraction layer. I think it's silly at this point and if you agree then I will just back those changes out. Avltree would simply be a stand-alone container and nothing else would changed.